src/engine/SCons/Tool/jar.py

```diff
@@ -98,18 +98,30 @@ def Jar(env, target = None, source = [], *args, **kw):
     # jar target should not be a list so assume they passed
     # no target and want implicit target to be made and the arg
     # was actaully the list of sources
-    if SCons.Util.is_List(target):
+    if SCons.Util.is_List(target) and source == []:
```

Comment:
In general all values returned from builders are NodeLists. So it's possible a user could specify a target as a list, but that list could have only one item in it. Perhaps better to also check if the list has more than one item here?

Reply:
Never mind. Your solution seems reasonable given a target list with one or more targets.